agent/command/assume_ec2_role.go

```diff
@@ -70,22 +68,42 @@ func (r *ec2AssumeRole) Execute(ctx context.Context,
 		return errors.WithStack(err)
 	}
 
+	// TODO (DEVPROD-9947): Remove feature flag check and just use the new implementation.
 	if r.TemporaryFeatureFlag {
 		return r.execute(ctx, comm, logger, conf)
 	}
 
 	return r.legacyExecute(ctx, comm, logger, conf)
 }
 
-func (r *ec2AssumeRole) execute(ctx context.Context,
-	comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
-	// TODO (DEVPROD-9945): Migration to new implementation.
-	return errors.New("temporary feature flag is enabled")
+func (r *ec2AssumeRole) execute(ctx context.Context, comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
+	request := apimodels.AssumeRoleRequest{
+		RoleARN: r.RoleARN,
+	}
+	if r.DurationSeconds > 0 {
+		request.DurationSeconds = utility.ToInt32Ptr(r.DurationSeconds)
+	}
+	if r.Policy != "" {
+		request.Policy = utility.ToStringPtr(r.Policy)
+	}
+	td := client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}
+	creds, err := comm.AssumeRole(ctx, td, request)
+	if err != nil {
+		return errors.Wrap(err, "assuming role")
+	}
+	if creds == nil {
+		return errors.New("nil credentials returned")
+	}
+	conf.NewExpansions.PutAndRedact(globals.AWSAccessKeyId, creds.AccessKeyID)
+	conf.NewExpansions.PutAndRedact(globals.AWSSecretAccessKey, creds.SecretAccessKey)
+	conf.NewExpansions.PutAndRedact(globals.AWSSessionToken, creds.SessionToken)
+	conf.NewExpansions.PutAndRedact(globals.AWSRoleExpiration, creds.Expiration)
```

Comment:
I don't have strong opinions, but does the expiration time need to be redacted as well?

Reply:
No, that is hilarious. I didn't even think about it- I'll remove the redacting for that one